Sources/Provider/ProvideItemRequestStateController.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  ProvideItemRequestStateController.swift
+//  Provider
+//
+//  Created by Twig on 8/30/22.
+//  Copyright © 2022 Lickability. All rights reserved.
+//
+
+import Foundation
+import Networking
+import Persister
+import Combine
+
+/// A class responsible for representing the state and value of a provider item request being made.
+public final class ProvideItemRequestStateController<Item: Providable> {
+    
+    /// The state of a provider request's lifecycle.
+    public enum ProvideItemRequestState {
+        
+        /// A request that has not yet been started.
+        case notInProgress
+        
+        /// A request that has been started, but not completed.
+        case inProgress
+        
+        /// A request that has been completed with an associated result.
+        case completed(Result<Item, ProviderError>)
+                
+        /// A `Bool` representing if a request is in progress.
+        public var isInProgress: Bool {
+            switch self {
+            case .notInProgress, .completed:
+                return false
+            case .inProgress:
+                return true
+            }
+        }
+        
+        /// The completed `LocalizedError`, if one exists.
+        public var completedError: LocalizedError? {
+            switch self {
+            case .notInProgress, .inProgress:
+                return nil
+            case let .completed(result):
+                switch result {
+                case .success:
+                    return nil
+                case let .failure(error):
+                    switch error {
+                    case let .networkError(networkError):
+                        return networkError
+                    case let .decodingError(error):
+                        if let decodingError = error as? DecodingError {
+                            print(decodingError)
+                        }
+                        
+                        return error as? LocalizedError
+                    default:
```

Comment:
Any chance I can get you to be exhaustive here (still `return error`) so newly added cases produce a compiler error?